components/suggestion/suggestion_date.tsx

```diff
@@ -1,20 +1,13 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import Popover from 'components/widgets/popover';
 
-export default class SuggestionDate extends React.PureComponent {
-    static propTypes = {
-        onCompleteWord: PropTypes.func.isRequired,
-        items: PropTypes.array.isRequired,
-        terms: PropTypes.array.isRequired,
-        components: PropTypes.array.isRequired,
-        matchedPretext: PropTypes.array.isRequired,
```

Comment:
lets define what all props are actually used in this component and declare the props interface with them